test/unit/specs/DefaultSlideTemplate.spec.js

```diff
@@ -2,272 +2,356 @@ import Vue from 'vue'
 import DefaultSlideTemplate from '@/components/Templates/DefaultSlideTemplate'
 
 describe('DefaultSlideTemplate.vue', function () {
-  let vm // <- component object to be redefined
-  const Constructor = Vue.extend(DefaultSlideTemplate)
-  /* definition of props to be passed down */
-  const props = {
-    carousel: 0,
-    slide: {
-      images: [
-        { src: 'https://www.usask.ca/programs/images/college.jpg' },
-        { src: 'http://csee2015.usask.ca/images/3611278449_e68f87520b_b.jpg' }
-      ],
-      title: {
-        content: 'Sherbrooke Event Title',
-        fontColor: 'Blue',
-        fontSize: 'Large',
-        fontStyle: 'Normal',
-        fontWeight: 'Oblique'
-      },
-      description: {
-        content: 'Sherbrooke event slide body. Lorem ipsum dolor sit amet, pri veniam forensibus id. Vis maluisset molestiae id ad semper lobortis cum. At impetus detraxit incorrupte usu, repudiare assueverit ex eum, ne nam essent vocent admodum.',
-        fontColor: 'Red',
-        fontSize: 'Small',
-        fontStyle: 'Normal',
-        fontWeight: 'Lighter'
-      },
-      date: {
-        content: '2018-03-07',
-        fontColor: 'Green',
-        fontSize: 'Medium',
-        fontStyle: 'Italic',
-        fontWeight: 'Bolder'
-      },
-      time: {
-        content: '11:05 AM',
-        fontColor: 'Yellow',
-        fontSize: 'Smaller',
-        fontStyle: 'Normal',
-        fontWeight: 'Bold'
-      },
-      meta: {
-        template: 'DefaultSlideTemplate',
-        timeout: '90',
-        repeatable: true,
-        startDate: '2018-07-25',
-        endDate: '2018-09-01'
-      }
-    }
-  }
+  let vm, Constructor // <- objects to be redefined
 
-  // reset the test case
-  beforeEach('Reset the contents to the preset style of props', function () {
-    // pass data into propsData
-    vm = new Constructor({ propsData: props })
+  before(function () {
+    Constructor = Vue.extend(DefaultSlideTemplate)
```

Comment:
the constructor can be hoisted to the scope of the spec as well, because this test only tests `DefaultSlideTemplate` and nothing else

Reply:
done